MetaMask/metamask-mobile
# fix: Warning when rejecting an approval request with id XYZ not found

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add the appropiate QA label when dev review is completed
    - `needs-qa`: PR requires manual QA.
    - `No QA/E2E only`: PR does not require any manual QA effort. Prior to merging, ensure that you have successful end-to-end test runs in Bitrise. 
    - `Spot check on release build`: PR does not require feature QA but needs non-automated verification. In the description section, provide test scenarios. Add screenshots, and or recordings of what was tested.
5. Add `QA Passed` label when QA has signed off (Only required if the PR was labeled with `needs-qa`)
6. Add your team's label, i.e. label starting with `team-` (or `external-contributor` label if your not a MetaMask employee)

**Description**

Whenever you deploy an ERC 20, create an approval request, move the app to the background and get it back active again.
Basically, the issue is that `handleAppStateChange` and `componentWillUnmount` are calling reject approval twice without checking if the approval exists.

Changes:
- Includes a check if an approval request exists before trying to reject.

**Screenshots/Recordings**

[bug-warning-approval-id-not-found.webm](https://github.com/MetaMask/metamask-mobile/assets/45455812/7caca741-8e12-4ea3-ae2a-ddeb8fe0e89b)


_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**To Reproduce**
Steps to reproduce the behaviour

- Go to test dapp
- Deploy an ERC20 token
- Click Approve
- Go in the background and get the app back "active"
- See warning

The expected behaviour is the transaction being rejected but no warning or error is shown.

**E2E**
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/8c4f9cff-ec1b-43ed-8c6f-776cfb46d06f

**Issue**

fixes https://github.com/MetaMask/metamask-mobile/issues/6860

**Checklist**
